app/models/operatingsystem.rb

```diff
@@ -313,6 +313,28 @@ def pxe_kernel_options(params)
     options
   end
 
+  def last_modified_id(media_provider)
+    @last_modified_id ||= begin
+      hash_source = []
+      boot_files_uri(media_provider).each do |url|
+        begin
+          logger.debug "Performing HTTP HEAD #{url}"
+          uri = URI(url)
+          Net::HTTP.start(uri.host, uri.port) do |http|
```

Comment:
Do we need to ask the smart- 
proxy to do this? The proxy currently downloads the file, does foreman core have access to it? I‘m thinking of firewalls etc.

Reply:
I just remembered there was an attempt so solve this in smart-proxy: https://github.com/theforeman/smart-proxy/pull/472

I actually like that this does not depend on `wget` anymore. But I agree, it's a bit overengineered. Maybe we can create something similar using [down](https://github.com/janko/down).

Reply:
Good point, however my thinking is - if it's a repo that in our current Katello flow:

* Upstream (repo, CDN, custom repo)
* Foreman Server (with Katello)
* Remote Smart Proxy

This actually requires Foreman Server to have access to upstream repos. It should not big deal implementing a proxy-proxy but this creates another complexity and I wonder if anyone wants this. Maybe let's create such an endpoint he moment user asks for it?

Reply:
We at dmTECH actually have such a use case, that's why I brought it up. In addition, we have to solve this for other (non-RHEL use cases as well).
If Foreman needs access to all media directly, I'd consider this a bug.

Why can't we simply implement a freshness check in the smart-proxy? This seems a lot less intrusive than this change.